components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManager.java

```diff
@@ -0,0 +1,438 @@
+                throw new ConflictException(error);
+            }
+            RoleBasicInfo roleBasicInfo = roleManagementService
+                    .addRole(role.getDisplayName(), role.getUsers(), role.getGroups(), role.getPermissions(), tenantDomain);
+
+            Role createdRole = new Role();
+            createdRole.setId(roleBasicInfo.getId());
+            String locationURI = SCIMCommonUtils.getSCIMRoleURL(roleBasicInfo.getId());
+            createdRole.setLocation(locationURI);
+            createdRole.setDisplayName(roleBasicInfo.getName());
+            createdRole.setSchemas();
+
+            return createdRole;
+        } catch (IdentityRoleManagementException e) {
+            throw new CharonException(
+                    String.format("Error occurred while adding a new role: %s", role.getDisplayName()), e);
+        }
+    }
+
+    @Override
+    public Role getRole(String roleID, Map<String, Boolean> requiredAttributes)
+            throws BadRequestException, CharonException {
+
+        try {
+            org.wso2.carbon.identity.role.mgt.core.Role role = roleManagementService.getRole(roleID, tenantDomain);
+            Role scimRole = new Role();
+            scimRole.setId(role.getId());
+            scimRole.setDisplayName(role.getName());
+            String locationURI = SCIMCommonUtils.getSCIMRoleURL(role.getId());
+            scimRole.setLocation(locationURI);
+            scimRole.setPermissions(role.getPermissions());
+            scimRole.setSchemas();
+
+            if (CollectionUtils.isNotEmpty(role.getUsers())) {
+                for (UserBasicInfo userInfo : role.getUsers()) {
+                    String userLocationURI = SCIMCommonUtils.getSCIMUserURL(userInfo.getId());
+                    User user = new User();
+                    user.setUserName(userInfo.getName());
+                    user.setId(userInfo.getId());
+                    user.setLocation(userLocationURI);
+                    scimRole.setUser(user);
+                }
+            }
+            if (CollectionUtils.isNotEmpty(role.getGroups())) {
+                for (GroupBasicInfo groupInfo : role.getGroups()) {
+                    String groupLocationURI = SCIMCommonUtils.getSCIMGroupURL(roleID);
+                    Group group = new Group();
+                    group.setDisplayName(groupInfo.getName());
+                    group.setId(groupInfo.getId());
+                    group.setLocation(groupLocationURI);
+                    scimRole.setGroup(group);
+                }
+            }
+            return scimRole;
+
+        } catch (IdentityRoleManagementException e) {
+            throw new CharonException(String.format("Error occurred while getting the role: %s", roleID), e);
+        }
+    }
+
+    @Override
+    public void deleteRole(String roleID) throws CharonException {
+
+        try {
+            roleManagementService.deleteRole(roleID, tenantDomain);
```

Comment:
validate whether role exists and return not found exception if not.

Reply:
Fixed and updated the PR.